### PR TITLE
CSM Workshop Results

### PR DESCRIPTION
**Pull request by shundhammer**
Added features and requirements doc and current code status doc

### CODE REVIEW

File: doc/status-current-code.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+- Ugly:
+    - monolithic code with very long functions
+    - probably lots of code duplication over the years
+    - largely untestable
+    - relies heavily on "target map":
+        - nested lists and maps containing all kinds of storage-related data
+        - very difficult to debug anything
+        - undocumented
+
```
Comment (ancorgs) [inline doc/status-current-code.md]: NP: Nothing relevant, but we have different criteria for bad and ugly. :wink: For me the YCP leftovers are simply ugly, but lack of testability is clearly bad.

File: doc/yast-storage-requirements.md [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+    - the first logical partition is always /dev/sd..5 (no matter if /dev/sd..4 etc. exist)
+
+- GPT  (GUID Partition Table; new PCs with UEFI boot support)
+    - can be used on all supported architectures: i586, x86_64, ppc, s/390
+    - large number of partitions (128 or more)
+    - no longer limited by available space in MBR
+    - partitions keep their numbers when other partitions are removed or added
+    - parted will use the smallest available number when creating a new partition
+
+- Other:
```
Comment (ancorgs) [inline doc/yast-storage-requirements.md]: The organization of the content got me a bit lost. What about?
```
### Partitions table types supported by YaST Storage

- MS-DOS (...)
    - Blah
    - Blah
- GPT (...)
    - Other blah
    - More blah
- DASD (s/390 specific)
    - Detailed explanation missing
    - By the way

### Not supported or very limited support

Support for these types may be dropped in the future.

- BSD
    - Blah
- Sun
```

File: doc/status-current-code.md (hunk 1 above)
Comment (shundhammer) [inline doc/status-current-code.md]: We (Arvin and I) were considering the "ugly" category worse than the "bad" category. ;-)

File: doc/yast-storage-requirements.md [hunk 3]
```diff
@@ -0,0 +1,293 @@
+# YaST Storage Features and Requirements #
```
Comment (ancorgs) [inline doc/yast-storage-requirements.md]: All along the document it's not so clear what is an existing feature and what is a (new?) requirement.

**Comment by ancorgs** [comment]
Not the most structured documentation ever. But it's much better to have it into the repo in that form than to have it only in @shundhammer and @aschnell heads.

**Comment by kobliha** [comment]
And I've already asked for this several times before, so *thumbs up* :)

**Pull request merged**